clustered/client/src/main/java/org/ehcache/clustered/client/config/builders/ClusteringServiceConfigurationBuilder.java

```diff
@@ -90,18 +114,19 @@ public ServerSideConfigurationBuilder expecting() {
    *
    * @throws NullPointerException if {@code unit} is {@code null}
    * @throws IllegalArgumentException if {@code amount} is negative
+   *
+   * @deprecated Use {@link #timeouts(Timeouts)}. Note that calling this method will override any timeouts previously set
+   * by setting the read operation timeout to the specified value and everything else to its default.
    */
+  @Deprecated
   public ClusteringServiceConfigurationBuilder readOperationTimeout(long duration, TimeUnit unit) {
-    return new ClusteringServiceConfigurationBuilder(this, TimeoutDuration.of(duration, unit));
+    Duration readTimeout = Duration.of(duration, toChronoUnit(unit));
```

Comment:
This can now throw, something that was not the case before. I believe we should hide that and simply use the largest `java.time.Duration` in case what is passed in is too large, as in the end, it will not make a difference.

Reply:
throw an exception you mean? Why do you care? It is deprecated anyway and the `Duration` will throw a nice exception telling what's wrong anyway.

Reply:
I care because it did not previously, so it is a behaviour change, much more sneaky than an API change. So like I said, I would catch it and use the largest duration we can represent.

Reply:
I found it much sneakier to modify the value set by the user :-(

Reply:
After discussion, I will move anything bigger than `Duration.ofNanos(Long.MAX_VALUE)` to `Duration.ofNanos(Long.MAX_VALUE)` in the `Timeouts` constructor

Reply:
I don't like this solution but I think it's the best way forward now.  I'll push for what I'd like to see on the expiry changes PR.